framework/wazuh/core/cluster/common.py

```diff
@@ -1047,6 +1055,334 @@ def get_node(self):
+        """
+        pivoting = self.get_payload != {} and self.pivot_key != ''
+        status = ''
+        chunks = []
+        last_pivot_value = 0
+        if pivoting:
+            self.get_payload[self.pivot_key] = last_pivot_value
+
+        try:
+            # Retrieve information from local wazuh-db
+            get_chunks_start_time = datetime.datetime.utcnow().timestamp()
+            while status != 'ok':
+                command = self.get_data_command + json.dumps(self.get_payload)
+                result = self.data_retriever(command=command)
+                status = result[0]
+                chunks.append(result[1])
+                if pivoting:
+                    try:
+                        last_pivot_value = json.loads(result[1])[-1]['data'][-1]['id']
+                        self.get_payload[self.pivot_key] = last_pivot_value
+                    except (IndexError, KeyError):
+                        pass
+        except exception.WazuhException as e:
+            self.logger.error(f"Error obtaining data from wazuh-db: {e}")
+            return []
+
+        self.logger.debug(f"Obtained {len(chunks)} chunks of data in "
+                          f"{(datetime.datetime.utcnow().timestamp() - get_chunks_start_time):.3f}s.")
+        return chunks
+
+    async def sync(self, start_time: float, chunks: List):
+        """Start sending information to master/worker node.
+
+        Parameters
+        ----------
+        start_time : float
+            Start time to be used when logging task duration if master/worker's response is not expected.
+        chunks : list
+            Data gathered from the database.
+
+        Returns
+        -------
+        bool
+            True if data was correctly sent to the master/worker node, None otherwise.
+        """
+        if chunks:
+            # Send list of chunks as a JSON string
+            data = json.dumps({'set_data_command': self.set_data_command,
+                               'payload': self.set_payload, 'chunks': chunks}).encode()
+            task_id = await self.manager.send_string(data)
+            if task_id.startswith(b'Error'):
+                raise exception.WazuhClusterError(3016, extra_message=f'String with agents information could '
+                                                                      f'not be sent to the master node: {task_id}')
+
+            # Specify under which task_id the JSON can be found in the master/worker.
+            await self.manager.send_request(command=self.cmd, data=task_id)
+            self.logger.debug(f"All chunks sent.")
+        else:
+            self.logger.info(
+                f"Finished in {(datetime.datetime.utcnow().timestamp() - start_time):.3f}s (0 chunks sent).")
+        return True
+
+
+def end_sending_agent_information(logger, start_time, response) -> Tuple[bytes, bytes]:
+    """Function called when the master/worker sends the "syn_m_a_e", "syn_m_g_e" or "syn_w_g_e" command.
```

Comment:
This will also be solved in the issue of the worker nodes, right now we are looking for this PR worker.py to work without the need to modify it: https://github.com/wazuh/wazuh/issues/11792